[review] MD-4: Postconfirmation: Gas Costs and Rewards (#4)
movementlabsxyz/MIP: MD/md-4/rollover-gas.md

# Rollover Gas
The current suggestion for the implementation of MCR is to offset the gas costs for the last attester. This would be done transparently, by rewarding the last attester for their work in additional compensation for this gas cost. This would be a simple solution to the problem of disincentivizing the last attester. However, this solution is naive as it may incentivize an attempt to be the last attester--introducing an incentive for latency to the chain.

## Base Model
We model the utility of the offset as follows...

```math
\begin{align}
&U_{n} = -E[G_{\mu}] \\
&U_{l} = \delta(|\mathcal{V}|)U_{o} - E[G_{\mathcal{V}}] \\
\end{align}
```
$U_{n}$ is the utility of any attester that is not last.

$U_{l}$ is the utility of the last attester.

> 

Reviewer: ```suggestion

$E [.] is the expectation value of a variable.$

```

$E[G]$ is the expected gas cost modeled as a random variable $G$.

$\delta(|\mathcal{V}|)$ is reward rate determined by the number of validators in the set $\mathcal{V}$.

Note that $G \not\!\perp\!\!\!\perp \delta(|\mathcal{V}|)$, as the gas cost is a function of the number of validators in the set. This will be addressed in later sections.

For an attester with complete information not to attempt to be last, $E[U_{n}] > E[U_{l}]$ which would imply that the $\delta(|\mathcal{V}|) < g(|\mathcal{V}|)$ where $g$ the gas algebra applied be Ethereum. 

## Incomplete Information
However, an attester would not have complete information about whether the would be last. In general, this is unknowable because of FLP impossibility.

To model this, we will say an attester can play two strategies $L$ and $N$ where $L$ is the strategy to be last and $N$ is the strategy to not be last. The utility of these strategies is given by...

```math
\begin{align}
& U_{N} = P[W = 0] U_{n} + P[W = 1] U_{l} + E[P_{N}] \\
& U_{L} = P[W = 0] U_{n} + P[W = 1] U_{l} + E[P_{L}] \\
\end{align}
```
$W$ is a random variable indicating whether the attester is last.

> $P_{N}$ is a random variable representing the price of attempting to commit as soon as possible. It can be implemented by the smart contract by giving greater rewards to those who commit earlier.

Reviewer: ```suggestion
$P_{N}$ is a random variable representing the price of attempting to commit as soon as possible. It can be implemented by considering that the commitment time is somewhat random under strategy $N$ (considering there are several actors utilizing strategy $N$), and by giving greater rewards to those who commit earlier. 
```

> $P_{L}$ is the price of the strategy $L$ which is the price of attempting to be last. It can fall under the same mechanism as the above, in which case its expectation can be computed over a narrower range of the same distribution as $P_{N}$.

Reviewer: ```suggestion
$P_{L}$ is a random variable, representing the price of the strategy $L$ which is the price of attempting to be last. It can fall under the same mechanism as the above, in which case its expectation can be computed over a narrower range of the same distribution as $P_{N}$.
```

Reviewer: - it is difficult to understand the problems as MD-4 is missing some context. I strongly recommend to give an introduction, otherwise readers will be very lost. One para in the overview would do.
- in particular it is not clear why the utility functions would be valid. 
- it is not clear why all attesters have only costs but no rewards (U_n) , while the last attester has a revenue (U_l)
- without a minimal description of the protocol it is not possible to understand why there is a liveness issue, or where it originates from, and why there is need for such a complicated equilibria game

Reviewer: In conclusion the issues that the model attempts to describe seem to be caused by the fact that there is a reward to being last. 

However this seems not desireable, causes questionable equilibria games, and overall raises the question of liveness problems. I would question whether such a design makes sense and if not a better design can be found.
